### PR TITLE
feat: [CURLRequest] add option for Proxy

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- add request option `proxy`

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: tests/system/HTTP/CURLRequestDoNotShareOptionsTest.php [hunk 1]
```diff
@@ -559,6 +559,20 @@ public function testSSLWithBadKey()
         ]);
     }
 
+    public function testProxyuOption()
+    {
+        $this->request->request('get', 'http://example.com', [
+            'proxy' => 'http://localhost:3128',
```
Comment (datamweb) [inline tests/system/HTTP/CURLRequestDoNotShareOptionsTest.php:565]: I think `proxy` should be an array that can set `CURLOPT_PROXY`, `CURLOPT_PROXYTYPE `, `CURLOPT_PROXYPORT`  and etc.

Comment (kenjis) [reply tests/system/HTTP/CURLRequestDoNotShareOptionsTest.php:565]: It seems Guzzle does not have such options.
https://docs.guzzlephp.org/en/stable/request-options.html#proxy

**Review by ddevsr — COMMENTED** [review]
Can flexible option with basic below?

```
[
    'proxy' => [
        'host' => '127.0.0.1:1337',
        'type' => 'sock5', // sock5 or sock4 or default with CURLPROXY_HTTP
        'auth' => 'user:pass', // or array
    ]
]
```

**Comment by datamweb** [comment]
> Can flexible option with basic below?

It seems not. Document **Guzzle** gives an example as below:
```php
$client->request('GET', '/', [
    'proxy' => [
        'http'  => 'http://localhost:8125', // Use this proxy with "http"
        'https' => 'http://localhost:9124', // Use this proxy with "https",
        'no' => ['.mit.edu', 'foo.com']    // Don't use a proxy with these
    ]
]);
```
However, I think the documents of this PR should clearly indicate how to set these things.

**Comment by kenjis** [comment]
@ddevsr 

> This class is modeled after the [Guzzle HTTP Client](http://docs.guzzlephp.org/en/latest/) library since it is one of the more widely used libraries. Where possible, the syntax has been kept the same so that if your application needs something a little more powerful than what this library provides, you will have to change very little to move over to use Guzzle.
https://codeigniter4.github.io/CodeIgniter4/libraries/curlrequest.html

I think CURLRequest is a lightweight Guzzle, and if you need more functionality, use Guzzle.
Therefore, we should avoid extensions of our own.

**Review by ddevsr — COMMENTED** [review]
Btw dont forget adding note for auth like http://username:password@127.0.0.1:10

**Comment by jozefrebjak** [comment]
@kenjis thanks for this. I really need this as soon as possible in my app.

**Comment by kenjis** [comment]
@ddevsr If someone needs it, s/he can implement and send a PR.

**Pull request merged**